src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -169,8 +169,29 @@ private static boolean hasJavadocComment(DetailAST methodDef) {
      * @return true if a token has a javadoc comment.
      */
     private static boolean hasJavadocCommentOnToken(DetailAST methodDef, int tokenType) {
+        boolean result = false;
+
         final DetailAST token = methodDef.findFirstToken(tokenType);
-        return token.branchContains(TokenTypes.BLOCK_COMMENT_BEGIN);
+        DetailAST curNode = token;
```

Comment:
please move it to static function.

Should it be util method ? sounds line very likely to be reused.

Reply:
@rnveach , ping.
Lets finish this PR :) .

Reply:
> please move it to static function

It already is in a static method.

> Should it be util method ?

How do you want to make this a util method? It has a result to assign, a condition to meet to assign the value, and an early end condition.

Once it is merged, `if (curNode.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {` has to be changed to fix https://github.com/checkstyle/checkstyle/issues/5450 .

Reply:
hasJavadocCommentOnToken is specific to this Check method.
you can make `SomeUtil.branchContains(token, TokenTypes.BLOCK_COMMENT_BEGIN);`

ok ... if you plan to refactor it further in knonw place, ok to merge as is.